wms_connector/models/synchronize_exportable_mixin.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2023 Akretion
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import base64
+import csv
+import datetime
+import uuid
+from io import StringIO
+
+from odoo import fields, models
+
+
+class SynchronizeExportableMixin(models.AbstractModel):
+    _name = "synchronize.exportable.mixin"
+    _description = "Synchronizable export mixin"
+    wms_export_date = fields.Date()
+    wms_export_attachment = fields.Many2one("attachment.queue", index=True)
+    wms_export_error = fields.Char()
+    file_creation_mode = fields.Selection(
```

Comment:
file_creation_mode is not a field on the mixing (you do not want to fill this info manually on every records)

Just a property in the class that return the file_creation_mode


```
@property
def file_creation_mode(self):
    return per_record
```

Then in the product export just inherit the class to change the value